### PR TITLE
feat: Add `caip25CaveatBuilder` to `@metamask/multichain`

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

Moves the validator from the `caip25EndowmentBuilder` into a new `caip25CaveatBuilder` helper to fix the incorrect full validation of the CAIP-25 caveat inside the CAIP-25 permission.

## References

See: https://github.com/MetaMask/metamask-extension/pull/27847#discussion_r1881175092
Extension: https://github.com/MetaMask/metamask-extension/pull/29166

### `@metamask/multichain`

- **BREAKING**: The validator returned by ` caip25EndowmentBuilder` now only verifies that there is single CAIP-25 caveat and nothing else.
- **ADDED**: Adds `caip25CaveatBuilder` helper that builds a specification for the CAIP-25 caveat that can be passed to the relevant `PermissionController` constructor param.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

**Comment by jiexi** [comment]
@metamaskbot publish-preview

**Review by adonesky1 — DISMISSED** [review]
LGTM!

File: packages/multichain/src/caip25Permission.test.ts [hunk 1]
```diff
@@ -394,14 +395,7 @@ describe('caip25EndowmentBuilder', () => {
...
   describe('permission validator', () => {
-    const findNetworkClientIdByChainId = jest.fn();
-    const listAccounts = jest.fn();
-    const { validator } = caip25EndowmentBuilder.specificationBuilder({
-      methodHooks: {
-        findNetworkClientIdByChainId,
-        listAccounts,
-      },
-    });
+    const { validator } = caip25EndowmentBuilder.specificationBuilder({});
```
Comment (Gudahtt) [inline packages/multichain/src/caip25Permission.test.ts:398]: Nit: Strange that it lets you pass in an empty object here 🤔 It looks like it should take no parameters?

```suggestion
    const { validator } = caip25EndowmentBuilder.specificationBuilder();
```

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by adonesky1 — APPROVED** [review]

File: packages/multichain/src/caip25Permission.test.ts (hunk 1 above)
Comment (Gudahtt) [reply packages/multichain/src/caip25Permission.test.ts]: Ah, the type _does_ require an options object, but we have it typed as `Record<never, never>`. Probably we should revert my suggestion here for now, but ultimately I think we should consider updating the specification builder type to let us exclude the options completely.

**Pull request merged**